src/Nethermind/Nethermind.Core/Eip2930/AccessList.cs

```diff
@@ -1,30 +1,75 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using System.Collections.Generic;
 using Nethermind.Int256;
 
-namespace Nethermind.Core.Eip2930
+namespace Nethermind.Core.Eip2930;
+
+public class AccessList
 {
-    public class AccessList
+    private readonly List<object> _items;
+
+    private AccessList(List<object> items)
+    {
+        _items = items;
+    }
+
+    public static AccessList Empty() => new(new List<object>());
+
+    public IEnumerable<(Address Address, IEnumerable<UInt256> StorageKeys)> AsEnumerable()
+    {
+        IEnumerable<UInt256> GetStorageKeys(int i)
+        {
+            while (i < _items.Count && _items[i] is UInt256 storageKey)
+            {
+                yield return storageKey;
+                i++;
+            }
+        }
+
+        for (int i = 0; i < _items.Count; i++)
+        {
+            object item = _items[i];
+            switch (item)
+            {
+                case Address address:
```

Comment:
on 2nd thought this could be simplified to:

`if (_items[i] is Address address)`